fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteSimpleAggToHDFSScanRule.java

```diff
@@ -0,0 +1,251 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class RewriteSimpleAggToHDFSScanRule extends TransformationRule {
+    public static final RewriteSimpleAggToHDFSScanRule HIVE_SCAN_NO_PROJECT =
+            new RewriteSimpleAggToHDFSScanRule(OperatorType.LOGICAL_HIVE_SCAN, true);
+    public static final RewriteSimpleAggToHDFSScanRule ICEBERG_SCAN_NO_PROJECT =
+            new RewriteSimpleAggToHDFSScanRule(OperatorType.LOGICAL_ICEBERG_SCAN, true);
+    public static final RewriteSimpleAggToHDFSScanRule FILE_SCAN_NO_PROJECT =
+            new RewriteSimpleAggToHDFSScanRule(OperatorType.LOGICAL_FILE_SCAN, true);
+
+    public static final RewriteSimpleAggToHDFSScanRule HIVE_SCAN =
+            new RewriteSimpleAggToHDFSScanRule(OperatorType.LOGICAL_HIVE_SCAN);
+    public static final RewriteSimpleAggToHDFSScanRule ICEBERG_SCAN =
+            new RewriteSimpleAggToHDFSScanRule(OperatorType.LOGICAL_ICEBERG_SCAN);
+    public static final RewriteSimpleAggToHDFSScanRule FILE_SCAN =
+            new RewriteSimpleAggToHDFSScanRule(OperatorType.LOGICAL_FILE_SCAN);
+
+    final OperatorType scanOperatorType;
+    final boolean hasProjectOperator;
+
+    private RewriteSimpleAggToHDFSScanRule(OperatorType logicalOperatorType, boolean withoutProject) {
+        super(RuleType.TF_REWRITE_SIMPLE_AGG, Pattern.create(OperatorType.LOGICAL_AGGR)
+                .addChildren(Pattern.create(logicalOperatorType)));
+        hasProjectOperator = false;
+        scanOperatorType = logicalOperatorType;
+    }
+
+    private RewriteSimpleAggToHDFSScanRule(OperatorType logicalOperatorType) {
+        super(RuleType.TF_REWRITE_SIMPLE_AGG, Pattern.create(OperatorType.LOGICAL_AGGR)
+                .addChildren(Pattern.create(OperatorType.LOGICAL_PROJECT, logicalOperatorType)));
+        hasProjectOperator = true;
+        scanOperatorType = logicalOperatorType;
+    }
+
+    private OptExpression buildAggScanOperator(LogicalAggregationOperator aggregationOperator,
+                                               LogicalScanOperator scanOperator,
+                                               OptimizerContext context) {
+        ColumnRefFactory columnRefFactory = context.getColumnRefFactory();
+        Map<ColumnRefOperator, CallOperator> aggs = aggregationOperator.getAggregations();
+
+        Map<ColumnRefOperator, CallOperator> newAggCalls = Maps.newHashMap();
+        Map<ColumnRefOperator, Column> newScanColumnRefs = Maps.newHashMap();
+
+        // select out partition columns.
+        int tableRelationId = -1;
+        for (ColumnRefOperator c : scanOperator.getColRefToColumnMetaMap().keySet()) {
+            int relationId = columnRefFactory.getRelationId(c.getId());
+            if (tableRelationId == -1) {
+                tableRelationId = relationId;
+            } else {
+                Preconditions.checkState(tableRelationId == relationId, "Table relation id is different across columns");
```

Comment:
I think this rule is for optimization, if it failed to transform the optExpression, we can give up this transformation. So maybe record a log instead of throwing exception to ensure the sql can be executed.

Reply:
OK. Will fix that.